python-package/setup.py

```diff
@@ -32,25 +31,31 @@ def find_lib():
 def copy_files(use_gpu=False):
 
     def copy_files_helper(folder_name):
-        src = os.path.join('..', folder_name)
+        src = os.path.join(CURRENT_DIR, os.path.pardir, folder_name)
         if os.path.exists(src):
-            dst = os.path.join('./compile', folder_name)
+            dst = os.path.join(CURRENT_DIR, 'compile', folder_name)
             shutil.rmtree(dst, ignore_errors=True)
             distutils.dir_util.copy_tree(src, dst)
         else:
             raise Exception('Cannot copy {} folder'.format(src))
 
-    if not os.path.isfile('./_IS_SOURCE_PACKAGE.txt'):
+    if not os.path.isfile(os.path.join(CURRENT_DIR, '_IS_SOURCE_PACKAGE.txt')):
         copy_files_helper('include')
         copy_files_helper('src')
-        if not os.path.exists("./compile/windows/"):
-            os.makedirs("./compile/windows/")
-        distutils.file_util.copy_file("../windows/LightGBM.sln", "./compile/windows/LightGBM.sln")
-        distutils.file_util.copy_file("../windows/LightGBM.vcxproj", "./compile/windows/LightGBM.vcxproj")
+        if not os.path.exists(os.path.join(CURRENT_DIR, "compile", "windows")):
+            os.makedirs(os.path.join(CURRENT_DIR, "compile", "windows"))
+        distutils.file_util.copy_file(os.path.join(CURRENT_DIR, os.path.pardir, "windows", "LightGBM.sln"),
+                                      os.path.join(CURRENT_DIR, "compile", "windows", "LightGBM.sln"))
+        distutils.file_util.copy_file(os.path.join(CURRENT_DIR, os.path.pardir, "windows", "LightGBM.vcxproj"),
+                                      os.path.join(CURRENT_DIR, "compile", "windows", "LightGBM.vcxproj"))
+        distutils.file_util.copy_file(os.path.join(CURRENT_DIR, os.path.pardir, "windows", "LightGBM.vcxproj.filters"),
+                                      os.path.join(CURRENT_DIR, "compile", "windows", "LightGBM.vcxproj.filters"))
```

Comment:
@guolinke Do we need to copy this file too?

Refer to https://github.com/Microsoft/LightGBM/commit/7689a4d0647b07cc3357e28e3eb85ae24d7e919a

Reply:
I think `filters` is not needed for build